shell/platform/linux/fl_mouse_cursor_plugin.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_mouse_cursor_plugin.h"
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_method_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_standard_method_codec.h"
+
+#include <gtk/gtk.h>
+
+static constexpr char kChannelName[] = "flutter/mousecursor";
+static constexpr char kBadArgumentsError[] = "Bad Arguments";
+static constexpr char kActivateSystemCursorMethod[] = "activateSystemCursor";
+static constexpr char kKindKey[] = "kind";
+
+struct _FlMouseCursorPlugin {
+  GObject parent_instance;
+
+  FlMethodChannel* channel;
+
+  FlView* view;
+};
+
+G_DEFINE_TYPE(FlMouseCursorPlugin, fl_mouse_cursor_plugin, G_TYPE_OBJECT)
+
+// Sets the mouse cursor.
+FlMethodResponse* activate_system_cursor(FlMouseCursorPlugin* self,
+                                         FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_MAP) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument map missing or malformed", nullptr));
+  }
+
+  FlValue* kind_value = fl_value_lookup_string(args, kKindKey);
+  const gchar* kind = nullptr;
+  if (fl_value_get_type(kind_value) == FL_VALUE_TYPE_STRING)
+    kind = fl_value_get_string(kind_value);
+
+  const gchar* cursor_name = nullptr;
+  if (g_strcmp0(kind, "none") == 0)
+    cursor_name = "none";
+  else if (g_strcmp0(kind, "basic") == 0)
+    cursor_name = "default";
+  else if (g_strcmp0(kind, "click") == 0)
+    cursor_name = "pointer";
+  else if (g_strcmp0(kind, "text") == 0)
+    cursor_name = "text";
+  else if (g_strcmp0(kind, "forbidden") == 0)
+    cursor_name = "not-allowed";
+  else if (g_strcmp0(kind, "grab") == 0)
+    cursor_name = "grab";
+  else if (g_strcmp0(kind, "grabbing") == 0)
+    cursor_name = "grabbing";
+  else
+    cursor_name = "default";
+
+  GdkWindow* window =
+      gtk_widget_get_window(gtk_widget_get_parent(GTK_WIDGET(self->view)));
```

Comment:
Why do you have to use the parent to get the window rather than getting the window of the view directly?

Reply:
Should have been `gtk_widget_get_toplevel`.